onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -46,8 +54,89 @@ def __init__(self, model_path, data_reader: CalibrationDataReader, calibrate_op_
         self.white_nodes = white_nodes
         self.augmented_model_path = augmented_model_path
         self.input_name_to_nodes = {}
+        self.calibration_cache = {} # save temporary calibration table
+
+
+    def set_data_reader(self, data_reader):
+        self.data_reader = data_reader
+
+    def write_calibration_table(self):
```

Comment:
>   def write_calibration_table(self): [](start = 1, length = 37)

this function is very specific. it should not be in this function.

Reply:
https://github.com/rmccorm4/tensorrt-utils/blob/20.01/classification/imagenet/ImagenetCalibrator.py#L166

Here is an example from TensorRT utils repo and its Calibrator class includes write_calibration_table() as well.

Reply:
emm, it is an implementation of IInt8EntropyCalibrator2, and i don't think they have same design as us.
Its functionality is to dump the parameters to a file. Why not use a util function to implement it? 

---
In reply to: [541832986](https://github.com/microsoft/onnxruntime/pull/6103#discussion_r541832986) [](ancestors = 541832986)

Reply:
I've moved this function to quant_utils.py